spring-integration-ip/src/main/java/org/springframework/integration/ip/config/IpAdapterParserUtils.java

```diff
@@ -150,7 +153,40 @@ public static void addConstuctorValueIfAttributeDefined(BeanDefinitionBuilder bu
 	 * @param builder The builder.
 	 * @param parserContext The parser context.
 	 */
-	public static void addHostAndPortToConstructor(Element element,
+	public static void addBasicSocketConfigToConstructor(Element element,
+			BeanDefinitionBuilder builder, ParserContext parserContext) {
+		String socketExpression = element.getAttribute(IpAdapterParserUtils.SOCKET_EXPRESSION);
+		if (StringUtils.hasText(socketExpression)) {
+			addSocketExpressionToConstructor(socketExpression, element, builder, parserContext);
+		} else {
```

Comment:
Minor, of course, but that is something used to make some noise during review/merge:
the `else`/`catch`/`finally` on new line.
Please, take a look to the document: https://github.com/spring-projects/spring-framework/wiki/Spring-Framework-Code-Style.

As well as any affected classes should be updated in the `Copyright` to the current year.
I can fix everything that on merge, if I don't find any critical concerns.